### DISCUSSION TITLE
Global scope

### DISCUSSION THREAD

**Discussion by kijung-iM** [web-infra-dev/rspress#1211] [Q&A]
Hello, 

reference: https://docusaurus.io/docs/markdown-features/react#mdx-component-scope

I wonder if there is a way to apply component scripts globally in rspress like docusaurus. I looked through the documentation but couldn't find it.

**Comment by Timeless0911** [comment]
https://rspress.dev/api/config/config-basic#globaluicomponents  see this

**Reply by kijung-iM** [reply]
Thanks, 

I've tried it a few times, but it doesn't seem to work. Does the `globalUIComponents` option also work with mdx files?

For example, when I write the Status.js file as below and set it to the `globalUIComponents` option,

```js
import React from 'react';

export default function Status ({children}) {
  return (
    <code
      className={children}>
      {children}
    </code>
  )
}
```

Could not load component from mdx file.

**Reply by Timeless0911** [reply]
I know what you mean, you should use https://rspress.dev/api/config/config-build#markdownglobalcomponents instead.

**Reply by kijung-iM** [reply]
Good!! Thanks a lot!

**Comment by kijung-iM** [comment]
@Timeless0911 

Can you tell me one more thing?

https://rspress.dev/api/client-api/api-components

Is there a way to use internal components globally as well?

**Reply by Timeless0911** [reply]
No, you should import them in where you use them.
